### PR TITLE
BigQuery query log crawler [sc-12584]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

fetch audit logs from big query for server side processing

### 🤓 What?

- fetch audit logs from big query

### 🧪 Tested?

tested locally against BIgQuery instance.

### CODE REVIEW

File: metaphor/bigquery/README.md [hunk 1]
```diff
@@ -101,6 +101,27 @@ The max number of concurrent requests to the google cloud API can be configured
...
+  # (Optional) Number of days of query logs to fetch. Default to 1. If 0, the no query logs will be fetched.
+  lookback_days: <days>
+  
+  # (Optional) A list of users whose queries will be excluded from the log fetching.
+  excluded_usernames:
+    - <user_name1>
+    - <user_name2>
+
+  # (Optional) Exclude queries issued by service accounts. Default to true.
+  exclude_service_accounts: <boolean>
```
Comment (mars-lan) [inline metaphor/bigquery/README.md:119]: Probably want to set this to false by default

File: metaphor/bigquery/README.md [hunk 2]
```diff
@@ -101,6 +101,27 @@ The max number of concurrent requests to the google cloud API can be configured
...
+
+  # (Optional) The number of query logs to fetch from BigQuery in one batch. Default to 1000.
```
Comment (mars-lan) [inline metaphor/bigquery/README.md:121]: I believe max is also 1000?

Comment (alyiwang) [reply metaphor/bigquery/README.md:121]: I didn't find documentation on the max page size. But 1000 seems a reasonable value that works in other bq crawlers.

File: metaphor/bigquery/README.md (hunk 1 above)
Comment (alyiwang) [reply metaphor/bigquery/README.md:119]: updated

File: metaphor/bigquery/README.md (hunk 2 above)
Comment (mars-lan) [reply metaphor/bigquery/README.md:121]: There is. Need to look in the rest API doc

**Review by mars-lan — APPROVED** [review]

**Pull request merged**